### PR TITLE
:0 should go to first line

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CC = cc
+CC = gcc
```
Comment (mattn) [inline Makefile:1]: Ooops. I'll revert this soon.

**Comment by kyx0r** [comment]
Use :1 ?

**Comment by mattn** [comment]
:0 should lso work.

**Comment by aligrudi** [comment]
mattn ***@***.***> wrote:
> :0 should lso work.

I think :0 is an error according to ex(1p).  However, at the moment
commands like :0p and :0r do not work and your patches solves this
issue; I shall apply it.

I shall also apply the following patch to make ":2r !cmd" work.

Please report if you experience any problems.

Thanks,
Ali

diff --git a/ex.c b/ex.c
index 244c33f..e1790ee 100644
--- a/ex.c
+++ b/ex.c
@@ -359,15 +359,15 @@ static int ec_edit(char *loc, char *cmd, char *arg)
 static int ec_read(char *loc, char *cmd, char *arg)
 {
 	char msg[128];
-	int beg, end;
+	int beg, end, pos;
 	char *path;
 	char *obuf;
 	int n = lbuf_len(xb);
 	path = arg[0] ? arg : ex_path();
 	if (ex_region(loc, &beg, &end))
 		return 1;
+	pos = lbuf_len(xb) ? end : 0;
 	if (arg[0] == '!') {
-		int pos = MIN(xrow + 1, lbuf_len(xb));
 		char *ecmd = ex_pathexpand(arg, 1);
 		if (!ecmd)
 			return 1;
@@ -378,7 +378,6 @@ static int ec_read(char *loc, char *cmd, char *arg)
 		free(obuf);
 	} else {
 		int fd = open(path, O_RDONLY);
-		int pos = lbuf_len(xb) ? end : 0;
 		if (fd < 0) {
 			ex_show("read failed\n");
 			return 1;

**Comment by mattn** [comment]
Traditional vi http://ex-vi.sourceforge.net/ works :0.

**Comment by mattn** [comment]
This PR already seems to be merged.

**Comment by polluks** [comment]
@mattn Indeed, just close this PR.

**Pull request closed without merging**